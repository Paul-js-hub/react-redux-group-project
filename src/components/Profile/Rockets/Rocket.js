import PropTypes from 'prop-types';
import Card from 'react-bootstrap/Card';

const Rockets = (props) => {
  const { name } = props;

  return (
    <Card body>
      {name}
    </Card>
  );
};

Rockets.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Rockets;
