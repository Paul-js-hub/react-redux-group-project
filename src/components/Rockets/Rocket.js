import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import style from './rockets.module.css';

const Rocket = (props) => {
  const {
    id, image, name, desc, handleClick, reserved,
  } = props;

  return (
    <Row className="align-items-center my-5">
      <Col xl={3} md={5}>
        <div className="rocketImage">
          <img src={image} alt={name} width="100%" />
        </div>
      </Col>
      <Col>
        <div className={style.content}>
          <h2>{name}</h2>
          <p className={style.description}>
            {reserved && (<Badge className={style.badge} bg="success">Reserved</Badge>)}
            {desc}
          </p>
        </div>
        {/* {reserveBtn(reserved)} */}
        <Button
          type="button"
          onClick={() => handleClick(id)}
          variant={reserved ? ('secondary') : ('primary')}
          className={style.btn}
        >
          {reserved ? ('Cancel Reservation') : ('Reserve Rocket')}
        </Button>
      </Col>
      {/* {reserved ? ('hello')} */}
    </Row>
  );
};

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Rocket;
