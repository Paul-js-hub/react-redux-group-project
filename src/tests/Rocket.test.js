import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rocket from '../components/Rockets/Rockets';

describe('One rocket test', () => {
  it('Rocket render correctly', () => {
    const RocketPage = renderer.create(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    ).toJSON();
    expect(RocketPage).toMatchSnapshot();
  });

  test('Render rocket component', () => {
    render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
  });
});
