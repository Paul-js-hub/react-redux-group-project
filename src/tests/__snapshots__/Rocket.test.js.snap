// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One rocket test Rocket render correctly 1`] = `
<div
  className="container"
/>
`;
